core/include/scipp/core/variable.h

```diff
@@ -500,6 +500,8 @@ class SCIPP_CORE_EXPORT Variable {
 // The name should be changed to makeVariable after refactoring:
 // getting rid of all other makeVariable.
 template <class T, class... Ts> Variable makeVariable(Ts &&... ts) {
+  static_assert(!std::disjunction_v<std::is_lvalue_reference<Ts>...>,
+                "makeVariable inputs must be r-value references");
```

Comment:
@OwenArnold I did not think enough about this at the time, this addition is now breaking some sensible uses. Do you remember why it was added?

Reply:
It can be removed. I remember getting some weird errors on accidentally passing l-value references, but I don't particularly like this fix anyway.